File: mo-common/mo-common-mybatis/src/main/java/ink/ckx/mo/common/mybatis/annotation/DataPermission.java
package ink.ckx.mo.common.mybatis.annotation;

import java.lang.annotation.*;

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/21
 */
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
public @interface DataPermission {

    /**
     * 数据权限 {@link com.baomidou.mybatisplus.extension.plugins.inner.DataPermissionInterceptor}
     */
    String deptAlias() default "";

    String deptIdColumnName() default "dept_id";

    String userAlias() default "";

    String userIdColumnName() default "id";
}

File: mo-common/mo-common-mybatis/src/main/kotlin/ink/ckx/mo/common/mybatis/annotation/DataPermission.kt
package ink.ckx.mo.common.mybatis.annotation

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/21
 */
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(
    AnnotationTarget.ANNOTATION_CLASS,
    AnnotationTarget.CLASS,
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER
)
annotation class DataPermission(

    /**
     * 数据权限 [com.baomidou.mybatisplus.extension.plugins.inner.DataPermissionInterceptor]
     */
    val deptAlias: String = "",

    val deptIdColumnName: String = "dept_id",

    val userAlias: String = "",

    val userIdColumnName: String = "id"
)